packages/common/src/store/reducers.ts

```diff
@@ -312,6 +319,9 @@ export type CommonState = {
     publishPlaylistConfirmationModal: PublishPlaylistConfirmationModalState
     mobileOverflowModal: MobileOverflowModalState
     modals: ModalsState
+    modalsWithState: {
```

Comment:
would the plan be to have models just all unified at some point?

Reply:
yeah this could use some discussion. We have other modals just toplevel here, so we could do the same and remove this nesting.

could probably also preemptively merge into `modals` by moving existing modals from `ModalName: boolean | 'closing'` to `ModalName: { isOpen: boolean | 'closing' }` 🤔  Should be a straightforward migration, just some types and underlying store structure, all the call sites stay the same